perl/cpan/3rdparty/BUILD.bazel

```
@@ -0,0 +1,13 @@
+load("//perl/cpan:perl_cpan_compiler.bzl", "perl_cpan_compiler")
+
+exports_files([
+    "cpanfile",
+    "cpanfile.snapshot",
+])
+
+perl_cpan_compiler(
```

How about
```suggestion
cpan_compile(
```
Analogous to rules_python/rules_uv's `pip_compile`.
Also, less smurfing :)

---

What is `smurfing`? I went with `perl_*` prefixes because I didn't know what `cpan` was until starting on this effort. I also definitely think `pip_compile` is not the right name, it should have been `pip_compiler` at least since what Bazel does is produce an executable target that is capable of compiling, but it doesn't do the compile itself. Thus, the thing being made should be called a `compiler`.

---

Maybe `cpan_compilation` would be the right name? After all this rule is not the compiler itself, its setting up a specific file to be compiled 🤔

---

I think of the target as the compiler. I'm not particularly married to the name but I do prefer the rules follow a noun naming convention vs a verb or adjective. Rules make "things" and targets are the instance of a "thing". As long as this sentiment holds then I'm a happy camper. Are there any other names you prefer with that in mind?